src/main/java/org/jabref/logic/importer/fetcher/DOABFetcher.java

```diff
@@ -0,0 +1,129 @@
+                BibEntry entry = JsonToBibEntry(metadataArray);
+                return Collections.singletonList(entry);
+            }
+            // multiple results
+            List<BibEntry> entries = new ArrayList<>(response.length());
+            for (int i = 0; i < response.length(); i++) {
+                JSONArray metadataArray = response.getJSONObject(i).getJSONArray("metadata");
+                BibEntry entry = JsonToBibEntry(metadataArray);
+                entries.add(entry);
+            }
+            return entries;
+        };
+    }
+
+    private BibEntry JsonToBibEntry(JSONArray metadataArray) {
+        BibEntry entry = new BibEntry();
+        List<Author> authorsList = new ArrayList<>();
+        List<Author> editorsList = new ArrayList<>();
+        StringJoiner keywordJoiner = new StringJoiner(",");
+        for (int i = 0; i < metadataArray.length(); i++) {
+            JSONObject dataObject = metadataArray.getJSONObject(i);
+            switch (dataObject.getString("key")) {
+                case "dc.contributor.author" -> authorsList.add(toAuthor(dataObject.getString("value")));
+                case "dc.type" -> entry.setField(StandardField.TYPE,
+                        dataObject.getString("value"));
+                case "dc.date.issued" -> entry.setField(StandardField.YEAR, String.valueOf(
+                        dataObject.getInt("value")));
+                case "oapen.identifier.doi" -> entry.setField(StandardField.DOI,
+                        dataObject.getString("value"));
+                case "dc.title" -> entry.setField(StandardField.TITLE,
+                        dataObject.getString("value"));
+                case "oapen.pages" -> entry.setField(StandardField.PAGES, String.valueOf(
+                        dataObject.getInt("value")));
+                case "dc.description.abstract" -> entry.setField(StandardField.ABSTRACT,
+                        dataObject.getString("value"));
+                case "dc.language" -> entry.setField(StandardField.LANGUAGE,
+                        dataObject.getString("value"));
+                case "publisher.name" -> entry.setField(StandardField.PUBLISHER,
+                        dataObject.getString("value"));
+                case "dc.identifier.uri" -> entry.setField(StandardField.URI,
+                        dataObject.getString("value"));
+                case "dc.subject.other" -> keywordJoiner.add(dataObject.getString("value"));
+                case "dc.contributor.editor" -> editorsList.add(toAuthor(dataObject.getString("value")));
+            }
+        }
+        entry.setField(StandardField.AUTHOR, toAuthorList(authorsList));
+        entry.setField(StandardField.EDITOR, toAuthorList(editorsList));
+        entry.setField(StandardField.KEYWORDS, String.valueOf(keywordJoiner));
+        return entry;
+    }
+
+    private Author toAuthor(String author) {
```

Comment:
What happens if you have multiple authors? 
https://directory.doabooks.org/handle/20.500.12854/79348?show=full 
And I think you can use AuthorList.parse () as the names are already in bibtex format

Reply:
> What happens if you have multiple authors? https://directory.doabooks.org/handle/20.500.12854/79348?show=full And I think you can use AuthorList.parse () as the names are already in bibtex format

I created a list from the Author class and all authors are added to it then converted to an AuthorList I will add a test case for multiple authors, it works the same as multiple editors there a test case provide for that, but I agree with you I think there is a better solution for that, can you describe what AuthorList.parse () actually does and how to use it, I am not sure I completely understand it.

Reply:
The nutshell interpretation of `AuthorList` is that it is intended to represent one or more authors written in bibtex format.

`AuthorList.parse` is intended to take a `String` consisting of one or more authors in [bibtex format](https://www.bibtex.com/f/author-field/), and store them internally in a `List` that you can access directly through `getAuthors()` if you need to.

For this case, I'd probably go with 
```
authorsList.addAll(
	AuthorList.parse(
		dataObject.getString("value"))
			.getAuthors())
```

even if we should only be getting one author from doab, and it isn't in bibtex format. Probably preprocess by checking for (Ed.) as in https://github.com/JabRef/jabref/pull/8598#discussion_r840825147

Is that explanation/example useful?

You seem to already have nailed down how to deal with output from AuthorList. Do you have any questions regarding it?

Reply:
Thanks, that was useful, and definitely a cleaner solution, I will implement this, and also start thinking about solutions to handle edgy cases like the one you mentioned here https://github.com/JabRef/jabref/pull/8598#discussion_r839941767 especially the author name that starts with Fatima which is formatted differently from the other authors.

Reply:
I also noticed something for some results the field for the number of pages can come in a different format like this result here https://directory.doabooks.org/rest/search?query=%22UAV%E2%80%90Based%20Remote%20Sensing%20Volume%202%22&expand=metadata
this can be problematic because the type of the return is not an integer anymore it is a string.